web/common/utils/formatter.ts

```diff
@@ -0,0 +1,63 @@
+type D3FormatFunc = (specifier: string) => (n: number | { valueOf(): number }) => string;
+
+export const getFormatter = (d3Format: D3FormatFunc, unit: string) => {
+  return (val: number): string => {
+    // Round to dismiss float imprecision
+    val = Math.round(val * 10000) / 10000;
+    switch (unit) {
+      case 'seconds':
+        return formatSI(d3Format, val, 's');
+      case 'bytes':
+      case 'bytes-si':
+        return formatDataSI(d3Format, val, 'B');
+      case 'bytes-iec':
+        return formatDataIEC(d3Format, val, 'B');
+      case 'bitrate':
+      case 'bitrate-si':
+        return formatDataSI(d3Format, val, 'bit/s');
+      case 'bitrate-iec':
+        return formatDataIEC(d3Format, val, 'bit/s');
+      default:
+        // Fallback to default SI scaler:
+        return formatDataSI(d3Format, val, unit);
+    }
```

Comment:
Can we remove bytes and bitrate cases or are you going to implement then later ?? In both cases this go to default.

Reply:
No it's intentional and it doesn't go to default, it continues on the following case (cf mangoes and papayas here https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/switch  )

Reply:
oh you are right, there is not a break or continue; I didn't see it sorry